assignment/assignment.hs

```diff
@@ -27,8 +29,13 @@ import Data.Char(intToDigit)
 -- might need to use `++` rather than `:` to build the result)
 
 -- myfilter :: (a -> Bool) -> [a] -> [a]
-
-
+checker func list x = 
+  if func x
+    then list++[x]
+  else
+    list  
+myfilter func x= foldl (checker func) [] x
+--i'm not really sure why i only have to pass one parameter into checker, but everytime i passed more i got errors.
 -------
```

Comment:
Looks good.

It might be more idiomatic to use a `where` clause

~~~ hs
myfilter :: (a -> Bool) -> [a] -> [a]
myfilter predicate  =
  foldl tester [] 
  where
    tester result value
      | predicate value = result ++ [ value ]
      | otherwise       = result
~~~

but your answer is the same code.